src/auth/auth.service.ts

```diff
@@ -58,46 +58,80 @@ export class AuthService {
       "expressSessionSecret",
     );
 
+    const logoutResult = await this.additionalLogoutTasks(req, logoutURL);
+
     if (expressSessionSecret) {
       req.logout(async (err) => {
         if (err) {
           // we should provide a message
           console.log("Logout error");
           console.log(err);
-          //res.status(HttpStatus.BAD_REQUEST);
         }
-        return await this.additionalLogoutTasks(req, logoutURL);
       });
+
+      return logoutResult;
     } else {
-      return await this.additionalLogoutTasks(req, logoutURL);
+      return logoutResult;
     }
-    if (logoutURL) {
-      return { logout: "successful", logoutURL: logoutURL };
-    }
-
-    return { logout: "successful" };
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  flattenObject = (obj: any) => {
+    const result: Record<string, unknown> = {};
+
+    for (const i in obj) {
+      if (typeof obj[i] === "object" && !Array.isArray(obj[i])) {
+        const temp = this.flattenObject(obj[i]);
+        for (const j in temp) {
+          result[i + "." + j] = temp[j];
+        }
+      } else {
+        result[i] = obj[i];
+      }
+    }
+    return result;
+  };
+
```

Comment:
I'd suggest move this function to the common/utility file

Reply:
@Junjiequan Thanks for the review and the good comment. It is ready now.